### PR TITLE
Feature/google auth

### PR DESCRIPTION
**Pull request by Baggerone**

### CODE REVIEW

File: application/auth/google/endpoint_legacy.go [hunk 1]
```diff
@@ -0,0 +1,13 @@
+// +build !go1.9
+
+package google
+
+import (
+	"golang.org/x/oauth2"
+)
+
+// Endpoint is Google's OAuth 2.0 endpoint.
+var Endpoint = oauth2.Endpoint{
```
Comment (Schparky) [inline application/auth/google/endpoint_legacy.go]: How does this not conflict with the same variable in `endpoint.go`?

File: application/models/organization.go [hunk 2]
```diff
@@ -55,10 +62,19 @@ func (o *Organization) ValidateUpdate(tx *pop.Connection) (*validate.Errors, err
...
 
 func (o *Organization) GetAuthProvider() (auth.Provider, error) {
+
 	if o.AuthType == AuthTypeSaml {
 		return saml.New([]byte(o.AuthConfig))
 	}
 
+	if o.AuthType == AuthTypeGoogle {
+		gCfg := google.GoogleConfig{}
+		json.Unmarshal([]byte(o.AuthConfig), &gCfg)
```
Comment (Schparky) [inline application/models/organization.go]: The SAML provider does the Unmarshal inside New. For consistency, should they be the same?

File: application/auth/google/endpoint_legacy.go (hunk 1 above)
Comment (Baggerone) [reply application/auth/google/endpoint_legacy.go]: I just borrowed those files, but it has to do with the top line `// +build !go1.9`

Comment (Schparky) [reply application/auth/google/endpoint_legacy.go]: Oh, I see. Thanks.

File: application/models/organization.go (hunk 2 above)
Comment (Baggerone) [reply application/models/organization.go]: Good point.

File: .env.example [hunk 3]
```diff
@@ -6,4 +6,11 @@ AWS_ACCESS_KEY_ID=
 AWS_SECRET_ACCESS_KEY=
 AWS_DEFAULT_REGION=us-east-1
 AWS_REGION=us-east-1
-ROLLBAR_TOKEN=
\ No newline at end of file
+
+# For Google OAuth
+GOOGLE_SESSION_SECRET=abc123
+
+AUTH_CALLBACK_URL=https://wecarry.org/auth/callback
```
[the remark this answers is not in the document]
Comment (Baggerone) [reply .env.example]: done

**Pull request merged**